docs/docs/gatsby-link.md

```diff
@@ -356,3 +356,19 @@ You can similarly check for file downloads:
 ```
 
 [egghead]: https://egghead.io/playlists/use-gatsby-s-link-component-to-improve-site-performance-and-simplify-site-development-7ed3ddfe
+
+## Limitations when appending query parameters or hash links programmatically
```

Comment:
General question - what about query parameters makes them untenable with the `Link` component?

This seems like a perfectly reasonable use-case, e.g.

```jsx
<Link to="/user?id=1234">Go to 1234 user</Link>
```

Is there more nuance here? Is it query param navigation on the _same_ page that doesn't work?

Reply:
Yes, it wouldn't work if you did `<Link to="?id=1234">Go to 1234 user</Link>`. For example, if I built a Tab control that used the query parameter for the current tab.

Reply:
Let's maybe dial back the language a bit then? 

Maybe something like:

```suggestion
## Recommendations for programmatic, in-app navigation
```